go/vt/vtgate/buffer/shard_buffer.go

```diff
@@ -561,6 +561,29 @@ func (sb *shardBuffer) stopBufferingLocked(reason stopReason, details string) {
 	go sb.drain(q, clientEntryError)
 }
 
+// interleaveExec calls f count / interval times, passing it a computed index
+// value. It's intended to be used to split a set of operations evenly over
+// an array of data in an "interleaved" fashion, as opposed to splitting the
+// array by contiguous ranges.
+func interleaveExec(interval, offset, count int, f func(int)) {
```

Comment:
Since it is going to be quite difficult to ensure that the drain actually happened in parallel, it maybe worth adding unit tests for this func with larger values than the max of 3 requests that `testBuffering1` creates.

Reply:
Btw, this is a pretty cool and interesting implementation of a super lightweight execution queue.